roles/provision_vm/tasks/main.yml

```diff
@@ -0,0 +1,65 @@
+---
+- name: Fail if provision_vm_cluster and provision_vm_esxi_hostname both weren't set or both were set
+  when: ( provision_vm_cluster is defined and provision_vm_esxi_hostname is defined ) or ( provision_vm_cluster is not defined and provision_vm_esxi_hostname is not defined) # noqa: yaml[line-length]
+  ansible.builtin.fail:
+    msg: provision_vm_esxi_hostname and provision_vm_cluster are mutually exclusive parameters
+
+- name: Create or update the VM
+  community.vmware.vmware_guest:
+    hostname: "{{ provision_vm_hostname }}"
+    username: "{{ provision_vm_username }}"
+    password: "{{ provision_vm_password }}"
+    validate_certs: "{{ omit if provision_vm_validate_certs is not defined else provision_vm_validate_certs }}"
```

Comment:
I think better to use `{{ provision_vm_validate_certs | d(omit) }}` it's less verbose.